modal/mount.py

```diff
@@ -331,93 +331,108 @@ async def _put_file(file_spec: FileUploadSpec) -> api_pb2.MountFile:
         logger.debug(f"Uploaded {len(uploaded_hashes)}/{n_files} files and {total_bytes} bytes in {time.time() - t0}s")
         return _MountHandle._from_id(resp.mount_id, resolver.client, None)
 
+    @staticmethod
+    def create_client_mount():
+        import modal
 
-Mount, AioMount = synchronize_apis(_Mount)
+        # Get the base_path because it also contains `modal_utils` and `modal_proto`.
+        base_path, _ = os.path.split(modal.__path__[0])
 
+        # TODO(erikbern): this is incredibly dumb, but we only want to include packages that start with "modal"
+        # TODO(erikbern): merge functionality with _function_utils._is_modal_path
+        prefix = os.path.join(base_path, "modal")
 
-def _create_client_mount():
-    # TODO(erikbern): make this a static method on the Mount class
-    import modal
+        def condition(arg):
+            return module_mount_condition(arg) and arg.startswith(prefix)
 
-    # Get the base_path because it also contains `modal_utils` and `modal_proto`.
-    base_path, _ = os.path.split(modal.__path__[0])
+        return _Mount.from_local_dir(base_path, remote_path="/pkg/", condition=condition, recursive=True)
+
+    @staticmethod
+    def _get_client_mount():
+        if config["sync_entrypoint"]:
+            return _Mount.create_client_mount()
+        else:
+            return _Mount.from_name(client_mount_name(), namespace=api_pb2.DEPLOYMENT_NAMESPACE_GLOBAL)
 
-    # TODO(erikbern): this is incredibly dumb, but we only want to include packages that start with "modal"
-    # TODO(erikbern): merge functionality with _function_utils._is_modal_path
-    prefix = os.path.join(base_path, "modal")
+    @staticmethod
+    def create_package_mounts(module_names: Sequence[str]) -> List["_Mount"]:
+        """Returns a `modal.Mount` that makes local modules listed in `module_names` available inside the container.
+        This works by mounting the local path of each module's package to a directory inside the container that's on `PYTHONPATH`.
+
+        **Usage**
+
+        ```python notest
+        import modal
+        import my_local_module
```

Comment:
Maybe it's just me, but having the `import` here makes the purpose of `create_package_mounts` less clear. If the user imports in global scope, you don't need the package mount as auto-mount will make it work anyway. 

To verify this I removed the `create_package_mounts` and put `my_local_module` on the PYTHONPATH:

```
PYTHONPATH="$HOME/xxx" MODAL_ENV=modal_labs modal run testy.py
✓ Initialized. View app at https://modal.com/apps/ap-AbemXJPADEIRCAB4ngMFO2
✓ Created objects.
├── 🔨 Created f.
├── 🔨 Created mount /home/ubuntu/modal/testy.py
└── 🔨 Created mount /home/ubuntu/xxx/my_local_module
did stuff
```

```python
# /home/ubuntu/xxx/my_local_module/__init__.py
def do_stuff():
    print("did stuff")
```

Reply:
yeah good point. let's remove the import statements from this comment!